modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EChain.java

```diff
@@ -46,6 +47,11 @@
     Type promote = null;
     Cast there = null;
     Cast back = null;
+    
+    /** Creates a new simple EChain */
```

Comment:
Maybe the comment should read "Creates a new rhs-only EChain" or something along those lines.